src/socketServer.ts

```diff
@@ -0,0 +1,71 @@
+import dotenv from 'dotenv';
+import cors from 'cors';
+import express, { Express } from 'express';
+import { createServer, Server as HTTPServer } from "http";
+import { ClientToServerEvents, ServerToClientEvents } from './models/events.model';
+import { Server as SocketServer } from 'socket.io';
+
+/**
+ * Keeps track of application lifecycle and maintains a testable server context.
+ *
+ * Not all fields or methods should be exposed as the application context. If a field is not marked as `private`, document the reason.
+ */
+export class socketServer {
+    private readonly serverContext: HTTPServer;
+
+    private readonly port: string;
+
+    constructor() {
+        // Load environment variables
+        dotenv.config({ path: `${__dirname}/../config/.env` });
+        this.port = process.env.SOCKET_SERVER_PORT || "8900";
+
+        const expressApp: Express = express();
+        expressApp.use(cors());
+
+        this.serverContext = createServer(expressApp);
+
+        this.socketServerInit(this.serverContext);
+
+        this.serverContext.listen(this.port, async () => {
+            // tslint:disable-next-line:no-console
+            console.log(`started server at http://localhost:${this.port}`);
+        });
+    }
+
+    /**
+     * Not private since tests will be required to confiugre their own instance of a @type {serverApplication}
```

Comment:
Do we want unit testing on the socket server? Haven't built one, so I can't speak for how testing would look.

Also it looks like `{serverApplication}` is leftover documentation.

Reply:
If we do want testing, btw, just copy this file over to enable autimation. It should work without any modification required: https://github.com/Project-SAP/project-sap-backend/blob/main/.github/workflows/node.js.yml